index.js

```diff
@@ -36,12 +36,37 @@ module.exports = function (options) {
 
   function validateState() {
     var isValid = state && (hasBounds() || state.isMaximized || state.isFullScreen);
+    if (!isValid) {
+      state = null;
```

Comment:
I'm fine with this for now. Will need to look into some general refactoring anyways.